frontend/dockerfile/docs/reference.md

```diff
@@ -818,6 +828,246 @@ The cache for `RUN` instructions can be invalidated by [`ADD`](#add) and [`COPY`
+| `mode`              | File mode for socket in octal. Default `0600`.                                                 |
+| `uid`               | User ID for socket. Default `0`.                                                               |
+| `gid`               | Group ID for socket. Default `0`.                                                              |
+
+#### Example: access to Gitlab
+
+```dockerfile
+# syntax=docker/dockerfile:1
+FROM alpine
+RUN apk add --no-cache openssh-client
+RUN mkdir -p -m 0700 ~/.ssh && ssh-keyscan gitlab.com >> ~/.ssh/known_hosts
+RUN --mount=type=ssh \
+  ssh -q -T git@gitlab.com 2>&1 | tee /hello
+# "Welcome to GitLab, @GITLAB_USERNAME_ASSOCIATED_WITH_SSHKEY" should be printed here
+# with the type of build progress is defined as `plain`.
+```
+
+```console
+$ eval $(ssh-agent)
+$ ssh-add ~/.ssh/id_rsa
+(Input your passphrase here)
+$ docker buildx build --ssh default=$SSH_AUTH_SOCK .
+```
+
+You can also specify a path to `*.pem` file on the host directly instead of `$SSH_AUTH_SOCK`.
+However, pem files with passphrases are not supported.
+
+## RUN --network
+
+> **Note**
+>
+> Added in [`docker/dockerfile:1.1`](#syntax)
+
+`RUN --network` allows control over which networking environment the command
+is run in.
+
+Syntax: `--network=<TYPE>`
+
+### Network types
+
+| Type                                         | Description                            |
+|----------------------------------------------|----------------------------------------|
+| [`default`](#run---networkdefault) (default) | Run in the default network.            |
+| [`none`](#run---networknone)                 | Run with no network access.            |
+| [`host`](#run---networkhost)                 | Run in the host's network environment. |
+
+### RUN --network=default
+
+Equivalent to not supplying a flag at all, the command is run in the default
+network for the build.
+
+### RUN --network=none
+
+The command is run with no network access (`lo` is still available, but is
+isolated to this process)
+
+#### Example: isolating external effects
+
+```dockerfile
+# syntax=docker/dockerfile:1
+FROM python:3.6
+ADD mypackage.tgz wheels/
+RUN --network=none pip install --find-links wheels mypackage
+```
+
+`pip` will only be able to install the packages provided in the tarfile, which
+can be controlled by an earlier build stage.
+
+### RUN --network=host
+
+The command is run in the host's network environment (similar to
+`docker build --network=host`, but on a per-instruction basis)
+
+> **Warning**
+>
+> The use of `--network=host` is protected by the `network.host` entitlement, 
+> which needs to be enabled when starting the buildkitd daemon with
+> `--allow-insecure-entitlement network.host` flag or in [buildkitd config](https://github.com/moby/buildkit/blob/master/docs/buildkitd.toml.md),
+> and for a build request with [`--allow network.host` flag](https://docs.docker.com/engine/reference/commandline/buildx_build/#allow).
```

Comment:
Could we import some of the buildx docs into this repo as a follow-up? Seems strange to me that we don't document entitlements in buildkit.